docroot/modules/custom/nvli_custom/nvli_custom.module

```diff
@@ -0,0 +1,29 @@
+<?php
+
+/**
+ * @file
+ * Contains nvli_custom.module.
+ */
+
+/**
+ * Implements hook_mirador_metadata_alter.
+ */
+function nvli_custom_mirador_metadata_alter(&$entity) {
+  // field_solr_doc_id contains the solr doc id, corresponding
+  // to this entity.
+  $solr_doc_id = $entity->field_solr_doc_id->value;
+  // @to-do: Get the solr server name from variable.
```

Comment:
Nope.. Even if we have multiple SOLR cores, the solr doc id will always be unique across all SOLR cores. Also, just replacing the service call with the one mentioned in the comment below would take care of everything.